### PR TITLE
[stdlib] Remove _ArrayProtocol as it's not adding much value

### PR DESCRIPTION
**Pull request by airspeedswift**
Since #19687 turns out to be a bit of a performance non-starter, this goes the other way and eliminates the protocol.

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by airspeedswift** [comment]
@swift-ci please smoke test

**Comment by airspeedswift** [comment]
@swift-ci please smoke test compiler performance

**Comment by airspeedswift** [comment]
@swift-ci please benchmark

File: test/api-digester/Outputs/stability-stdlib-source.swift.expected [hunk 1]
```diff
@@ -4,6 +4,9 @@
 /* RawRepresentable Changes */
 
 /* Removed Decls */
+Struct Array has removed conformance to _ArrayProtocol
```
[the remark this answers is not in the document]
Comment (nkcsgexi) [reply test/api-digester/Outputs/stability-stdlib-source.swift.expected]: That's a good point. A fix is committed to another [PR](https://github.com/apple/swift/pull/19744/commits/847aa9b3279926ecc9e4b41ea09cc879750939ba)

**Pull request closed without merging**